File: 171.乱序字符串/Solution.py
class Solution:
    # @param strs: A list of strings
    # @return: A list of strings
    def anagrams(self, strs):
        # write your code here
        d = {}
        for v in strs:
            v = ''.join(sorted(list(v)))
            if d.get(v, -1) == -1:
                d[v] = 1
            else:
                d[v] += 1
        rlt = [] 
        for v in strs:
            vm = ''.join(sorted(list(v)))
            if d[vm] > 1:
                rlt.append(v)
        return rlt

File: 171.乱序字符串/Solution.cpp
class Solution {  
public:      
    /** 
     * @param strs: A list of strings 
     * @return: A list of strings 
     */  
    vector<string> anagrams(vector<string> &strs) {  
        // write your code here  
        map<string, int> m;
        for(auto s:strs){
            sort(s.begin(), s.end());
            m[s]++;
        }
        
        vector<string> res;
        for(auto s:strs){
            auto temp = s;
            sort(temp.begin(), temp.end());
            if(m[temp] > 1)
                res.push_back(s);
                
        }
        
        return res;
    }  
};  
